archimedes/archimedes/runner.py

```diff
@@ -184,6 +185,7 @@ class LocalIsolator(Isolator[LocalProcess]):
 
     def is_running(self, t: LocalProcess) -> bool:
         p, io, f = t
+        p.poll()
```

Comment:
This fixes the main issue with the local executor, wherein essentially the python side does not update the returncode from the kernel process table without explicit polling here.  Tests were passing due to the short test timeout kicking in, which happened to cause the python process to update the process table, and pass, which is kind of hilarious.